public/javascripts/gmaps_google.js

```diff
@@ -0,0 +1,985 @@
+        }
+      }
+      if ((_ref = obj.included) != null) {
+        _ref.apply(this);
+      }
+      return this;
+    };
+
+    return Base;
+
+  })();
+
+}).call(this);
+(function() {
+  this.Gmaps.Objects.BaseBuilder = (function() {
+    function BaseBuilder() {}
+
+    BaseBuilder.prototype.build = function() {
+      return new (this.model_class())(this.serviceObject);
+    };
+
+    BaseBuilder.prototype.before_init = function() {};
+
+    BaseBuilder.prototype.after_init = function() {};
+
+    BaseBuilder.prototype.addListener = function(action, fn) {
+      return this.primitives().addListener(this.getServiceObject(), action, fn);
+    };
+
+    BaseBuilder.prototype.getServiceObject = function() {
+      return this.serviceObject;
+    };
+
+    BaseBuilder.prototype.primitives = function() {
+      return this.constructor.PRIMITIVES;
+    };
+
+    BaseBuilder.prototype.model_class = function() {
+      return this.constructor.OBJECT;
+    };
+
+    return BaseBuilder;
+
+  })();
+
+}).call(this);
+(function() {
+  this.Gmaps.Objects.Builders = function(builderClass, objectClass, primitivesProvider) {
+    return {
+      build: function(args, provider_options, internal_options) {
+        var builder;
+        objectClass.PRIMITIVES = primitivesProvider;
+        builderClass.OBJECT = objectClass;
+        builderClass.PRIMITIVES = primitivesProvider;
+        builder = new builderClass(args, provider_options, internal_options);
+        return builder.build();
+      }
+    };
+  };
+
+}).call(this);
+(function() {
+  this.Gmaps.Objects.Handler = (function() {
+    function Handler(type, options) {
+      this.type = type;
+      if (options == null) {
+        options = {};
+      }
+      this.setPrimitives(options);
+      this.setOptions(options);
+      this._cacheAllBuilders();
+      this.resetBounds();
+    }
+
+    Handler.prototype.buildMap = function(options, onMapLoad) {
+      var _this = this;
+      if (onMapLoad == null) {
+        onMapLoad = function() {};
+      }
+      return this.map = this._builder('Map').build(options, function() {
+        _this._createClusterer();
+        return onMapLoad();
+      });
+    };
+
+    Handler.prototype.addMarkers = function(markers_data, provider_options) {
+      var _this = this;
+      return _.map(markers_data, function(marker_data) {
+        return _this.addMarker(marker_data, provider_options);
+      });
+    };
+
+    Handler.prototype.addMarker = function(marker_data, provider_options) {
+      var marker;
+      marker = this._builder('Marker').build(marker_data, provider_options, this.marker_options);
+      marker.setMap(this.getMap());
+      this.clusterer.addMarker(marker);
+      return marker;
+    };
+
+    Handler.prototype.addCircles = function(circles_data, provider_options) {
+      var _this = this;
+      return _.map(circles_data, function(circle_data) {
+        return _this.addCircle(circle_data, provider_options);
```

Comment:
Identifier 'circle_data' is not in camel case.<br>Identifier 'provider_options' is not in camel case.